repose-aggregator/components/filters/header-translation-filter/src/test/scala/org/openrepose/filters/headertranslation/HeaderTranslationFilterTest.scala

```diff
@@ -174,22 +188,98 @@ class HeaderTranslationFilterTest extends FunSpec with BeforeAndAfter with Match
 
       filter.doFilter(request, null, mockFilterChain)
 
-      val requestCaptor = ArgumentCaptor.forClass(classOf[HttpServletRequest])
-      verify(mockFilterChain).doFilter(requestCaptor.capture(), any[ServletResponse])
-
-      val capturedRequest = requestCaptor.getValue
-      capturedRequest.getHeaderNames.toSeq should contain theSameElementsAs request.getHeaderNames.toSeq
+      getCapturedRequest.getHeaderNames.toSeq should contain theSameElementsAs request.getHeaderNames.toSeq
     }
 
     it("should not remove the original headers") {
       filter.doFilter(mockRequest, null, mockFilterChain)
 
-      val requestCaptor = ArgumentCaptor.forClass(classOf[HttpServletRequest])
-      verify(mockFilterChain).doFilter(requestCaptor.capture(), any[ServletResponse])
-
-      val capturedRequest = requestCaptor.getValue
+      val capturedRequest = getCapturedRequest
       capturedRequest.getHeaders("X-Three").toSeq should contain theSameElementsAs Seq("valueOne")
       capturedRequest.getHeaders("X-Four").toSeq should contain theSameElementsAs Seq("valueOne", "valueTwo")
     }
   }
+
+  describe("splittable header configuration") {
+    it("should not split the headers when NOT configured to do so") {
+      val mockRequest = new MockHttpServletRequest()
+      mockRequest.addHeader("X-Six", "mustard,ketchup,mayo;q=0.9,pickles;q=0.1")
+
+      filter.doFilter(mockRequest, null, mockFilterChain)
+
+      getCapturedRequest.getHeaders("X-New-Six-One").toSeq should contain theSameElementsAs Seq("mustard,ketchup,mayo;q=0.9,pickles;q=0.1")
+    }
+
+    it("should split the headers when configured") {
+      val mockRequest = new MockHttpServletRequest()
+      mockRequest.addHeader("X-Five", "mustard,ketchup,mayo;q=0.9,pickles;q=0.1")
+
+      filter.doFilter(mockRequest, null, mockFilterChain)
+
+      getCapturedRequest.getHeaders("X-New-Five-One").toSeq should contain theSameElementsInOrderAs Seq("mustard", "ketchup", "mayo;q=0.9", "pickles;q=0.1")
```

Comment:
You may be able to use the "inOrder" matcher rather than "theSameElementsInOrderAs". That would let you drop the Seq constructor. Or you could go a step further with the "inOrderOnly" matcher.

Again, this is of little consequence.